Mozu/core-theme
# Express Checkout with Default Payments and Visa Checkout Integration

### Feature: Express Checkout with Default Payments and Visa Checkout Integration

We enhanced the checkout experience and UI to allow an "Express Checkout" so that shoppers could check out quickly and easily with their pre-selected default payment and shipping options.

Users will see their default selections displayed in read-only format on the checkout page. Shoppers can configure their default selections in their My Account. To further enhance the express checkout capability, we’ve enabled integration with Visa's exclusive Visa Checkout digital wallet where shoppers can store their payment and shipping information. 
#### Additions
- We overhauled the `templates/modules/checkout/step-payment-info` template, which makes up the payment step of the default checkout flow. It now presents a different UI configuration, optimized for quick saved payment selection
- We modified several other Hypr templates in the checkout worfklow to enhance readability and usability.
- We overhauled the `scripts/modules/models-checkout` script to adhere to the new workflow and to deal differently with changes in payment details.
- We added Visa Checkout UI integrations to `templates/pages/cart` and the `templates/modules/checkout/step-payment-info` section, as well as additions to `scripts/pages/cart` and `scripts/pages/checkout` to incorporate the Visa Checkout integration behavior.
- We modified the My Account template `templates/modules/my-account/my-account-credit-card-form` to include Default Payment selection.
#### Integration Options

**The preferred upgrade path is always to update your theme to extend Core8.**

If doing so results in unrelated or unpredictable regressions, then instead you can inspect the changes we made to the Core Theme and manually integrate them.
#### Upgrade Process
- Extend Core8 or merge the files as described.
- Recompile your theme using the build tools.
- Perform regression tests.
- Test the functionality by:
  - Create a new customer. Visit My Account as that customer.. Create a payment method. Confirm that there is a new "Primary" selection. Select "primary" for the new payment method.
  - Add a product to a cart and proceed through a checkout. Confirm that the default payment method is already selected.
